### ISSUE TITLE
Agent.Monad missing

### ISSUE THREAD

**Issue by mor1** [yannham/mechaml#1]
Attempting to try this out, but `Agent.Monad` appears to be missing? cf.

```
: mort@greyjay:obapi#master % $; utop
-----------------------------------------+-------------------------------------------------------------+-----------------------------------------
                                         | Welcome to utop version 2.0.1 (using OCaml version 4.04.2)! |
                                         +-------------------------------------------------------------+
Findlib has been successfully loaded. Additional directives:
  #require "package";;      to load a package
  #list;;                   to list the available packages
  #camlp4o;;                to load camlp4 (standard syntax)
  #camlp4r;;                to load camlp4 (revised syntax)
  #predicates "p,q,...";;   to set these predicates
  Topfind.reset();;         to force that packages will be reloaded
  #thread;;                 to enable threads


Type #utop_help for help about using utop.

-( 22:12:49 )-< command 0 >-------------------------------------------------------------------------------------------------------{ counter: 0 }-
utop # #require "mechaml";;
-( 22:12:51 )-< command 1 >-------------------------------------------------------------------------------------------------------{ counter: 0 }-
utop # open Mechaml;;
-( 22:12:54 )-< command 2 >-------------------------------------------------------------------------------------------------------{ counter: 0 }-
utop # module M = Agent.Monad;;
Error: Unbound module Agent.Monad                                                                                                                -( 22:13:00 )-< command 3 >-------------------------------------------------------------------------------------------------------{ counter: 0 }-
utop # Mechaml.
+-----+---------+-----+----+--------------------------------------------------------------------------------------------------------------------+
|Agent|Cookiejar|Infix|Page|                                                                                                                    |
+-----+---------+-----+----+--------------------------------------------------------------------------------------------------------------------+
```

...should `Monad` be added to the relevant `Modules` line in the `_oasis`?

**Comment by yannham** [comment]
That's strange, I just tried it out on OCaml 4.05.0 (removed my local pin and installed mechaml from the opam repository) and it seems to be ok. I will try on other switches to see if that makes any difference. I'm quite a oasis newbie but I assumed submodules would be automatically exported too.

**Comment by yannham** [comment]
No problem on 4.04.02 either. How did you install Mechaml ?

**Comment by mor1** [comment]
Just `opam install mechaml`. 

`opam info mechaml` gives
```
             package: mechaml
             version: 0.1
          repository: upstream
        upstream-url: https://github.com/yannham/mechaml/archive/opam-release-0.1.tar.gz
       upstream-kind: http
   upstream-checksum: 9294b88d7fe4ce3b42b9de4363827681
            homepage: https://github.com/yannham/mechaml
         bug-reports: https://github.com/yannham/mechaml/issues
            dev-repo: git://github.com/yannham/mechaml.git
              author: Yann Hamdaoui <yann.hamdaoui@gmail.com>
             license: LGPL v3
             depends: ocamlbuild & ocamlfind & cohttp (>= 0.21.0 & < 0.99) & lwt & uri & lambdasoup < 0.7.0
   installed-version: 0.1 [system]
   available-version: 0.1
         description: A Mechanize-like web scraping library
Mechaml is a simple web scraping library that allows to :
* Fetch web content
* Analyze, fill and submit HTML forms
* Handle cookies, headers and redirections
* Use a web proxy **(soon to be implemented)**

Mechaml is built on top of existing libraries that alreay provide most of the
interesting features : Cohttp and Lwt for asynchronous I/O and HTTP handling,
and Lambdasoup to parse HTML. It provides an interface that handles the
interactions between these and add useful features.
```

`ocaml -version` reports `The OCaml toplevel, version 4.04.2`

I've double-checked that I ran `eval $(opam config env)` in the shell so paths should be correct; `opam switch` shows the active switch as `system  C system  System compiler (4.04.2)`

**Comment by mor1** [comment]
FWIW, `ls -l $(opam config var root)/system/lib/mechaml` gives:
```
total 1232
-rw-rw-r-- 1 mort staff    424 Mar 31 10:27 META
-rw-rw-r-- 1 mort staff  60818 Aug 30 05:53 agent.annot
-rw-rw-r-- 1 mort staff  68332 Aug 30 05:53 agent.cmt
-rw-rw-r-- 1 mort staff  34708 Aug 30 05:53 agent.cmti
-rw-rw-r-- 1 mort staff   4506 Mar 31 10:27 agent.mli
-rw-rw-r-- 1 mort staff  52699 Aug 30 05:53 cookiejar.annot
-rw-rw-r-- 1 mort staff  57642 Aug 30 05:53 cookiejar.cmt
-rw-rw-r-- 1 mort staff  24798 Aug 30 05:53 cookiejar.cmti
-rw-rw-r-- 1 mort staff   2683 Mar 31 10:27 cookiejar.mli
-rw-rw-r-- 1 mort staff 126928 Aug 30 05:53 mechaml.a
-rw-rw-r-- 1 mort staff 145604 Aug 30 05:53 mechaml.cma
-rw-rw-r-- 1 mort staff  21487 Aug 30 05:53 mechaml.cmi
-rw-rw-r-- 1 mort staff  23168 Aug 30 05:53 mechaml.cmt
-rw-rw-r-- 1 mort staff  20532 Aug 30 05:53 mechaml.cmx
-rw-rw-r-- 1 mort staff   5987 Aug 30 05:53 mechaml.cmxa
-rwxrwxr-x 1 mort staff 119700 Aug 30 05:53 mechaml.cmxs*
-rw-rw-r-- 1 mort staff      0 Aug 30 05:53 opam.config
-rw-rw-r-- 1 mort staff 201850 Aug 30 05:53 page.annot
-rw-rw-r-- 1 mort staff 167101 Aug 30 05:53 page.cmt
-rw-rw-r-- 1 mort staff  67039 Aug 30 05:53 page.cmti
-rw-rw-r-- 1 mort staff   8935 Mar 31 10:27 page.mli
```

**Comment by yannham** [comment]
The 0.1 version was the first one to be released and has no Agent.Monad module (and a questionnable API in general). The 1.0.0 version of Mechaml should be available for OCaml 4.02.xx. Could you try a `opam update` and see if you can upgrade Mechaml to 1.0.0 ?

**Comment by mor1** [comment]
Aha, doh!
Doing an update and upgrade has worked. Closing -- thanks!
